code/datums/status_effects/debuffs/terrified.dm

```diff
@@ -106,12 +106,13 @@
 				)
 			return COMPONENT_BLOCK_MISC_HELP
 
-	terror_buildup -= HUG_TERROR_AMOUNT
-	owner.visible_message(
-		span_notice("[owner] seems to relax as [hugger] gives [owner.p_them()] a comforting hug."),
-		span_nicegreen("You feel yourself calm down as [hugger] gives you a reassuring hug."),
-		span_hear("You hear shuffling and a sigh of relief."),
-	)
+	if(hugger != owner)
+		terror_buildup -= HUG_TERROR_AMOUNT
+		owner.visible_message(
+			span_notice("[owner] seems to relax as [hugger] gives [owner.p_them()] a comforting hug."),
+			span_nicegreen("You feel yourself calm down as [hugger] gives you a reassuring hug."),
+			span_hear("You hear shuffling and a sigh of relief."),
+		)
```

Comment:
A) early returns
B) do this before the nightmare check, in case something... weird, happens, and a nightmare gets terrified and scares the shit out of themselves.